lib/JSON2CSVBase.js

```diff
@@ -63,7 +64,7 @@ class JSON2CSVBase {
       : [row];
 
     if (this.opts.flatten) {
-      return processedRow.map(this.flatten);
+      return processedRow.map(this.flatten());
```

Comment:
This was adding a very unnecessary closure.

You could have accessed the right `this` but simple replacing `this.flatten` by `this.flatten.bind(this)` or `(row => this.flatten(row))`

Reply:
True, but that function is called only once, so it's really marginal. Feel free to submit a fix if it's gnawing at you 😉

Reply:
I would probably keep the function, and have it take an object of options.